src/vos/vos_aggregate.c

```diff
@@ -621,8 +626,12 @@ prepare_segments(struct agg_merge_window *mw)
 	 * segments (at most mw_lgc_cnt) and truncated segments (at most
 	 * mw_phy_cnt).
 	 */
-	D_ASSERT(mw->mw_lgc_cnt > 0);
+	D_ASSERT(mw->mw_lgc_cnt > 0 || mw->mw_rmv_cnt > 0);
 	D_ASSERT(mw->mw_phy_cnt > 0);
+	io->ic_seg_cnt = 0;
+	if (mw->mw_lgc_cnt == 0)
+		goto process_physical;
```

Comment:
hmm, why do we still needs to process physical entries? If there is no logical entry in the window, it means:
- Either this window has remove records only, or;
- The physical entry in this window part isn't visible (is covered by remove record);

In both cases, no physical entry needs be processed, right?

Reply:
We need to mark any such physical record for removal so that we call evt_delete on it.   See line 729

Reply:
indeed, I overlooked this case. This "process_physical" is before csum variable initialization, so I was thinking that new csum could be required for truncated physical record.